src/core/public/http/types.ts

```diff
@@ -217,30 +218,56 @@ export interface HttpFetchOptions extends HttpRequestInit {
   headers?: HttpHeadersInit;
 
   /**
-   * When `true` the return type of {@link HttpHandler} will be an {@link IHttpResponse} with detailed request and
+   * Whether or not the request should include the "system request" header to differentiate an end user request from
+   * Kibana internal request.
+   * Can be read on the server-side using KibanaRequest#isSystemRequest. Defaults to `false`.
+   */
+  asSystemRequest?: boolean;
+
+  /**
+   * When `true` the return type of {@link HttpHandler} will be an {@link HttpResponse} with detailed request and
    * response information. When `false`, the return type will just be the parsed response body. Defaults to `false`.
    */
   asResponse?: boolean;
 }
 
+/**
+ * Similar to {@link HttpFetchOptions} but with the URL path included.
+ * @public
+ */
+export interface HttpFetchOptionsWithPath extends HttpFetchOptions {
+  /*
+   * The path on the Kibana server to send the request to. Should not include the basePath.
+   */
+  path: string;
+}
+
 /**
  * A function for making an HTTP requests to Kibana's backend. See {@link HttpFetchOptions} for options and
- * {@link IHttpResponse} for the response.
+ * {@link HttpResponse} for the response.
  *
  * @param path the path on the Kibana server to send the request to. Should not include the basePath.
  * @param options {@link HttpFetchOptions}
- * @returns a Promise that resolves to a {@link IHttpResponse}
+ * @returns a Promise that resolves to a {@link HttpResponse}
  * @public
  */
 export interface HttpHandler {
+  /** @deprecated */
```

Comment:
wow. why we decided to deprecate it?

Reply:
Yeah maybe this is a bit heavy handed. I was thinking it would be nice if we had a single API, and the newer API that only accepts one object is what Jest  Mocks end up being typed as. That said, I'll take this off for now and leave that decision for another time.